examples/vision/yolov8.py

```diff
@@ -0,0 +1,628 @@
+)
+
+"""
+Next, let's build a YOLOV8 model using the `YOLOV8Detector`, which accepts a feature
+extractor as the `backbone` argument, a `num_classes` argument that specifies the number
+of object classes to detect based on the size of the `class_mapping` list, a
+`bounding_box_format` argument that informs the model of the format of the bbox in the
+dataset, and a finally, the feature pyramid network (FPN) depth is specified by the
+`fpn_depth` argument.
+
+It is simple to build a YOLOV8 using any of the aforementioned backbones thanks to
+KerasCV.
+
+"""
+
+yolo = keras_cv.models.YOLOV8Detector(
+    num_classes=len(class_mapping),
+    bounding_box_format="xyxy",
+    backbone=backbone,
+    fpn_depth=1,
+)
+
+"""
+## Compile the Model
+"""
+
+"""
+Loss used for YOLOV8
+
+
+1. Classification Loss: This loss function calculates the discrepancy between anticipated
+class probabilities and actual class probabilities. In this instance,
+`binary_crossentropy`, a prominent solution for binary classification issues, is
+Utilized. We Utilized binary crossentropy since each thing that is identified is either
+classed as belonging to or not belonging to a certain object class (such as a person, a
+car, etc.).
+
+2. Box Loss:  `box_loss` is the loss function used to measure the difference between the
+predicted bounding boxes and the ground truth. In this case, the Intersection over Union
+(IoU) metric is used, which measures the overlap between predicted and ground truth
+bounding boxes. Together, these loss functions help optimize the model for object
+detection by minimizing the difference between the predicted and ground truth class
+probabilities and bounding boxes.
+
+
+"""
+
+optimizer = tf.keras.optimizers.Adam(
+    learning_rate=LEARNING_RATE,
+    global_clipnorm=GLOBAL_CLIPNORM,
+)
+
+yolo.compile(
+    optimizer=optimizer, classification_loss="binary_crossentropy", box_loss="iou"
+)
+
+"""
+## COCO Metric Callback
+
+We will be using `BoxCOCOMetrics` from KerasCV to evaluate the model and calculate the
+Map(Mean Average Precision) score, Recall and Precision.
+"""
+
+
+class EvaluateCOCOMetricsCallback(keras.callbacks.Callback):
+    def __init__(self, data):
+        super().__init__()
+        self.data = data
+        self.metrics = keras_cv.metrics.BoxCOCOMetrics(
+            bounding_box_format="xyxy",
+            evaluate_freq=1e9,
+        )
+
+    def on_epoch_end(self, epoch, logs):
+        self.metrics.reset_state()
+        for batch in self.data:
+            images, y_true = batch[0], batch[1]
+            y_pred = self.model.predict(images, verbose=0)
+            self.metrics.update_state(y_true, y_pred)
+
+        metrics = self.metrics.result(force=True)
+        logs.update(metrics)
+        return logs
+
+
+"""
+## Train the Model
+"""
+
+yolo.fit(
+    train_ds,
+    validation_data=val_ds,
+    epochs=EPOCH,
+    callbacks=[EvaluateCOCOMetricsCallback(val_ds)],
```

Comment:
If possible, `callback.ModelCheckpoint` should be used here which will save the model by monitoring max `mAP`. 

Also. https://github.com/keras-team/keras-io/issues/1235

Reply:
Instead of using `callback.ModelCheckpoint` I have updated `EvaluateCOCOMetricsCallback` to save the model when mAP improves! :)